File: jpa-backend/src/main/java/chaincue/tech/jpabackend/masters/user_master/UserHelper.java
package chaincue.tech.jpabackend.masters.user_master;

import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
public class UserHelper {
    private final UserRepository userRepository;

    public UserHelper(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public User saveUser(User user) {
        return userRepository.save(user);
    }

    public Optional<User> findUser(String id) {
        return userRepository.findById(id);
    }

}

File: kotlin/jpa-backend/src/main/kotlin/chaincue/tech/jpabackend/masters/user_master/UserHelper.kt
package chaincue.tech.jpabackend.masters.user_master

import org.springframework.stereotype.Service
import java.util.*

@Service
class UserHelper(private val userRepository: UserRepository) {
    fun save(user: User): User = userRepository.save(user)
    fun findUser(id: String): Optional<User> = userRepository.findById(id)
}
